### PR TITLE
Update references, dependent variable

### PR DESCRIPTION
**Pull request by mounika-nalla**
### Description

As part of https://github.ibm.com/GoldenEye/issues/issues/4214 update hashicorp references 

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [ ] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@mounika-nalla In our most recent deep dive call, we agreed that we are going to stop bumping the required ibm provider version while updating dependant modules in our examples (more details in GoldenEye issue #4915). Instead going forward we are going to:

- Have 1 example always lock into the lowest version in the range (basic example that has little or no dependant modules)
- For other "complete" examples, do not lock into a specific version (or use range such as >= 1.49.0) so terraform will always pull the version that matches all dependant modules constraints.

I guess this is a good time to roll this out to this module so you don't have to change the required IBM provider version in the root level module.

Also I noticed that the readme also needs to be updated - it still pulls from github instead of hashicorp

File: examples/apply_taints/README.md [hunk 1]
```diff
@@ -12,7 +12,8 @@
 module "ocp_base" {
   # update this value to the value of your IBM Cloud API key
   ibmcloud_api_key     = "ibm cloud api key" # pragma: allowlist secret
-  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc.git?ref=master"
+  source  = "terraform-ibm-modules/base-ocp-vpc/ibm"
+  version = "latest" # Replace "latest" with a release version to lock into a specific release
```
Comment (ocofaigh) [inline examples/apply_taints/README.md:16]: This is actually a mistake, the source here should be `"../.."`

File: examples/add_rules_to_sg/version.tf [hunk 2]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Pin to the lowest provider version of the range defined in the main module to ensure lowest version still works
```
Comment (ocofaigh) [inline examples/add_rules_to_sg/version.tf:4]: this code comment needs to be removed now

File: examples/existing_cos/version.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Pin to the lowest provider version of the range defined in the main module to ensure lowest version still works
```
Comment (ocofaigh) [inline examples/existing_cos/version.tf:4]: this code comment needs to be removed now

File: examples/multiple_mzr_clusters/version.tf [hunk 4]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Pin to the lowest provider version of the range defined in the main module to ensure lowest version still works
```
Comment (ocofaigh) [inline examples/multiple_mzr_clusters/version.tf:4]: this code comment needs to be removed now

File: examples/single_zone_autoscale_cluster/version.tf [hunk 5]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Pin to the lowest provider version of the range defined in the main module to ensure lowest version still works
```
Comment (ocofaigh) [inline examples/single_zone_autoscale_cluster/version.tf:4]: this code comment needs to be removed now

File: examples/standard/version.tf [hunk 6]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Pin to the lowest provider version of the range defined in the main module to ensure lowest version still works
     ibm = {
       source  = "ibm-cloud/ibm"
-      version = "1.51.0"
+      version = ">= 1.51.0"
```
Comment (ocofaigh) [inline examples/standard/version.tf:7]: We need to keep one example locked into the lowest version supported by the module, so maybe lets keep this at `1.51.0`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see latest comments

**Comment by ocofaigh** [comment]
Due to the fact you are upgrading some modules used in the examples, they are now failing the upgrade test. Your going to have to skip it which is OK since there are no actual changes here to the root level module itself

File: examples/standard/version.tf (hunk 6 above)
Comment (ocofaigh) [reply examples/standard/version.tf:6]: @mounika-nalla still 1 open comment here

Comment (mounika-nalla) [reply examples/standard/version.tf:6]: This example uses the modules landing-zone-vpc (1.52.0) and key-protect-all-inclusive(1.53.0) which are set to latest versions of ibm provider. I left this example as it is now and modified the existing-cos example to lock into a specific version 1.51.0

**Comment by toddgiguere** [comment]
@ocofaigh We figured out why the PR test is failing due to consistency test reporting that the `ibm_container_addons.addons` requires an update following the plan (idempotent check).

The issue is with a change in the provider v1.53.0 and v1.54.0, probably part of this PR: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4408
When I upgrade the provider version of e-2-e example in `main` branch to v1.53.0, then follow with a plan, I get the following update from terraform:
```
  # module.ocp_all_inclusive.module.ocp_base.ibm_container_addons.addons will be updated in-place
  ~ resource "ibm_container_addons" "addons" {
        id                = "cicu0mmr021nefhes3dg"
      + manage_all_addons = true
      + managed_addons    = (known after apply)
        # (2 unchanged attributes hidden)

      + addons {
          + allowed_upgrade_versions = (known after apply)
          + deprecated               = (known after apply)
          + health_state             = (known after apply)
          + health_status            = (known after apply)
          + min_kube_version         = (known after apply)
          + min_ocp_version          = (known after apply)
          + name                     = "cluster-autoscaler"
          + options                  = (known after apply)
          + supported_kube_range     = (known after apply)
          + target_version           = (known after apply)
          + version                  = "1.0.8"
          + vlan_spanning_required   = (known after apply)
        }
      + addons {
          + allowed_upgrade_versions = (known after apply)
          + deprecated               = (known after apply)
          + health_state             = (known after apply)
          + health_status            = (known after apply)
          + min_kube_version         = (known after apply)
          + min_ocp_version          = (known after apply)
          + name                     = "vpc-block-csi-driver"
          + options                  = (known after apply)
          + supported_kube_range     = (known after apply)
          + target_version           = (known after apply)
          + version                  = "5.0"
          + vlan_spanning_required   = (known after apply)
        }

        # (1 unchanged block hidden)
    }
```

NOTE: when I try to `apply` that update, I actually get an error:
```
module.ocp_all_inclusive.module.ocp_base.ibm_container_addons.addons: Modifying... [id=cicu0mmr021nefhes3dg]
╷
│ Error: Request failed with status code: 409, ServerErrorResponse: {"incidentID":"c7bea872-cf41-429a-915e-8de4257b8039","code":"E4d8b","description":"An add-on with the same name already exists. To update your existing add-on to the latest version, see 'http://ibm.biz/addon-update'.","type":"Provisioning"}
│
│   with module.ocp_all_inclusive.module.ocp_base.ibm_container_addons.addons,
│   on .terraform/modules/ocp_all_inclusive.ocp_base/main.tf line 360, in resource "ibm_container_addons" "addons":
│  360: resource "ibm_container_addons" "addons" {
```

File: examples/apply_taints/README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@
 module "ocp_base" {
   # update this value to the value of your IBM Cloud API key
   ibmcloud_api_key     = "ibm cloud api key" # pragma: allowlist secret
-  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc.git?ref=master"
+  source  = "../.."
```
Comment (ocofaigh) [inline examples/apply_taints/README.md:15]: this should reference an actual source here, not a relative path

File: tests/pr_test.go [hunk 8]
```diff
@@ -52,6 +52,12 @@ func setupOptions(t *testing.T, prefix string, terraformDir string) *testhelper.
 			"ocp_version": ocpVersion1,
 			"access_tags": permanentResources["accessTags"],
 		},
+		IgnoreUpdates: testhelper.Exemptions{
+			List: []string{
+				// Workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4658
+				"module.ocp_base.ibm_container_addons.addons",
```
Comment (ocofaigh) [inline tests/pr_test.go:58]: No longer needed if we bump to 1.55.0

**Review by toddgiguere — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**